pkg/ring/kv/dynamodb/dynamodb.go

```diff
@@ -59,69 +66,65 @@ func newDynamodbKV(cfg Config, logger log.Logger) (dynamodbKV, error) {
 		return dynamodbKV{}, err
 	}
 
-	sess, err := session.NewSession()
+	awsCfg, err := config.LoadDefaultConfig(context.Background(),
+		config.WithRegion(cfg.Region),
+	)
```

Comment:
I think we might want to keep the check if cfg.Region is nil. cfg.region is optional and if empty the aws config should find from the env